highlighting/lib/src/extension/compiler_extensions.dart

```diff
@@ -14,8 +16,9 @@ void skipIfHasPrecedingDot(DomainRegexMatch match, Response response) {
 }
 
 void scopeClassName(Mode mode, [Mode? parent]) {
-  if (mode.className != null && mode.className != '') {
-    mode.scope = mode.className;
+  if ((mode.className != null && mode.className != '') ||
+      mode.getRawClassName() == preserveNullString) {
+    mode.scope = mode.className ?? preserveNullString;
     mode.className = null;
```

Comment:
1. Looks like the test for `''` is not in the original HighlightJS.
2. Try to simplify this.